domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointController.kt

```diff
@@ -0,0 +1,291 @@
+            ExplorationCheckpointNotFoundException(
+              "Checkpoint with the explorationId $explorationId was not found " +
+                "for profileId ${profileId.internalId}."
+            )
+          )
+      }
+  }
+
+  /**
+   * @return [ExplorationCheckpointDetails]  which contains the explorationId, explorationTitle
+   * and explorationVersion of the oldest saved checkpoint for the specified profile.
+   */
+  fun retrieveOldestSavedExplorationCheckpointDetails(
+    profileId: ProfileId
+  ): DataProvider<ExplorationCheckpointDetails> {
+    return retrieveCacheStore(profileId)
+      .transformAsync(
+        RETRIEVE_OLDEST_CHECKPOINT_DETAILS_DATA_PROVIDER_ID
+      ) { explorationCheckpointDatabase ->
+
+        // find the oldest checkpoint by timestamp or null if no checkpoints is saved.
+        val oldestCheckpoint = explorationCheckpointDatabase.explorationCheckpointMap.minByOrNull {
+          it.value.timestampOfFirstCheckpoint
+        }
+
+        if (oldestCheckpoint == null) {
+          AsyncResult.failed(
+            ExplorationCheckpointNotFoundException(
+              "No saved checkpoints in $CACHE_NAME for profileId ${profileId.internalId}."
+            )
+          )
+        } else {
+          val explorationCheckpointDetails = ExplorationCheckpointDetails.newBuilder()
+            .setExplorationId(oldestCheckpoint.key)
+            .setExplorationTitle(oldestCheckpoint.value.explorationTitle)
+            .setExplorationVersion(oldestCheckpoint.value.explorationVersion)
+            .build()
+          AsyncResult.success(explorationCheckpointDetails)
+        }
+      }
+  }
+
+  /** deletes the saved checkpoint for a specified explorationId and profileId. */
+  fun deleteSavedExplorationCheckpoint(
+    profileId: ProfileId,
+    explorationId: String
+  ): DataProvider<Any?> {
+    val deferred = retrieveCacheStore(profileId).storeDataWithCustomChannelAsync(
+      updateInMemoryCache = true
+    ) { explorationCheckpointDatabase ->
+
+      if (!explorationCheckpointDatabase.explorationCheckpointMap.containsKey(explorationId))
+        return@storeDataWithCustomChannelAsync Pair(
+          explorationCheckpointDatabase,
+          ExplorationCheckpointActionStatus.CHECKPOINT_NOT_FOUND
+        )
+
+      val explorationCheckpointDatabaseBuilder = explorationCheckpointDatabase.toBuilder()
+
+      explorationCheckpointDatabaseBuilder
+        .removeExplorationCheckpoint(explorationId)
+
+      Pair(
+        explorationCheckpointDatabaseBuilder.build(),
+        ExplorationCheckpointActionStatus.SUCCESS
+      )
+    }
+    return dataProviders.createInMemoryDataProviderAsync(
+      DELETE_EXPLORATION_CHECKPOINT_DATA_PROVIDER_ID
+    ) {
+      return@createInMemoryDataProviderAsync getDeferredResult(
+        deferred = deferred,
+        profileId = profileId,
+        explorationId = explorationId
+      )
+    }
+  }
+
+  private suspend fun getDeferredResult(
```

Comment:
```suggestion
  private suspend fun getExplorationCheckpointStatusResult(
```

Reply:
This naming makes the variable names consistent with other controllers.